### PR TITLE
create controller filer for userId

### PR DESCRIPTION
**Pull request by JosephThawn**
controller file for user Id, but we are not sure about update portion.

### CODE REVIEW

File: server/controllers/userId.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const User = require("../models/user");
+
+// Create Controller file for get, update and delte one user Joe and James
+
+//get one user
+const getOneuser = async (req, res, next) => {
+  try {
+    const oneUser = await User.findById(req.params.userId);
```
Comment (hjkmines) [inline server/controllers/userId.js]: lets call this `user`

File: server/controllers/userId.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    res.status(200).setHeader("Content-type", "application/json").json(oneUser);
+  } catch (err) {
+    res.status(404).send(`Can't get user data, ${err.message}`);
+    next(err);
+  }
+};
+
+const updateOneUser = async (req, res, next) => {
+  try {
+    const oneUpdate = await User.findByIdAndUpdate(
```
Comment (hjkmines) [inline server/controllers/userId.js]: lets call this `user`

File: server/controllers/userId.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  } catch {
+    res.status(404).send(`Can't delete user data, ${err.message}`);
+    next(err);
+  }
+};
+module.exports = {
+  getOneuser,
+  updateOneUser,
+  deleteOneUser,
+};
```
Comment (hjkmines) [inline server/controllers/userId.js]: lets move this file to the already existing file to user.js in the controllers file

**Review by hjkmines — CHANGES_REQUESTED** [review]

**Review by hjkmines — APPROVED** [review]
lgtm! approved.

**Review by vmaineng — APPROVED** [review]
Looks great!

File: server/controllers/users.js [hunk 4]
```diff
@@ -39,11 +39,51 @@ const deleteUsers = async (req, res, next) => {
     }
 }
 
-// Exports
+const getOneuser = async (req, res, next) => {
```
Comment (FunkyColdMedyna) [inline server/controllers/users.js:42]: is there a reason this is not fully camelCase "getOneuser" ,and the others follow camel case like "updateOneUser"?

**Comment by JosephThawn** [comment]
LGTM

File: server/controllers/users.js (hunk 4 above)
Comment (JosephThawn) [reply server/controllers/users.js:42]: It just define a variable using camel case for getOneUser and updateOneuser. you can give any name you want.

File: server/controllers/userId.js (hunk 1 above)
Comment (JosephThawn) [reply server/controllers/userId.js]: change getOneUser to user?

File: server/controllers/users.js [hunk 5]
```diff
@@ -39,11 +39,51 @@ const deleteUsers = async (req, res, next) => {
...
+    next(err);
+  }
+};
+
+const updateOneUser = async (req, res, next) => {
+  try {
+    const userUpdate = await User.findByIdAndUpdate(
+      req.params.userId,
+      { $set: req.body },
+      { new: true, runValidators: true }
```
Comment (hjkmines) [inline server/controllers/users.js:57]: can you change this to something to `req.params`

**Review by hjkmines — CHANGES_REQUESTED** [review]

**Review by hjkmines — APPROVED** [review]

**Comment by FunkyColdMedyna** [comment]
lgtm

**Review by FunkyColdMedyna — APPROVED** [review]

**Pull request merged**